### PR TITLE
feat(eslint-plugin-twenty-ts): added no hardcoded colors rule

### LINKED ISSUE #1591: Migrate no-hardcoded-colors rule from eslint-plugin-twenty to eslint-plugin-twenty-ts
## Context

We are maintaining a package `eslint-plugin-twenty` to add our own linter rule and help the community share the same standards. We have introduced a new `eslint-plugin-twenty-ts` package where rules can be written in typescript and tested. We need to migrate from `eslint-plugin-twenty` to `eslint-plugin-twenty-ts`

## Scope
This ticket is about moving the `no-hardcoded-colors` rule from `eslint-plugin-twenty` to `eslint-plugin-twenty-ts`. 
- take a look at `twenty/packages/eslint-plugin-twenty/rules/no-hardcoded-colors.js`
- create a new rule within eslint-plugin-twenty-ts that does the same thing
- write tests to cover the different case
- make sure that the linter is still detecting hardcoded colors in the codebase

### PR DESCRIPTION
**Pull request by akhill10**
fixes #1591

### CODE REVIEW

File: packages/eslint-plugin-twenty-ts/index.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
   rules: {
     "effect-components": require("./src/rules/effect-components"),
+    "no-hardcoded-colors": require("./src/rules/no-hardcoded-colors"),
```
Comment (charlesBochet) [inline packages/eslint-plugin-twenty-ts/index.ts:4]: let's also remove it from eslint-plugin-twenty

File: packages/eslint-plugin-twenty-ts/src/tests/all.spec.ts [hunk 2]
```diff
@@ -83,3 +84,24 @@ ruleTester.run("effect-components", effectComponentsRule, {
     },
   ],
 });
+
+ruleTester.run("no-hardcoded-colors", noHardcodedColorsRule, {
```
Comment (charlesBochet) [inline packages/eslint-plugin-twenty-ts/src/tests/all.spec.ts:88]: could we actually break this all.spec.ts file into one spec.ts by rule?

**Review by charlesBochet — COMMENTED** [review]
Looks good, added a two comments! Thank you :)

**Comment by charlesBochet** [comment]
Arf, just saw: https://github.com/twentyhq/twenty/pull/1618
As #1618 is treating all, let's merge #1618, feel free to work on any other good first issue (you can ping me on the issue if you need more detail, happy to give more context if needed). We are also creating good first issues regularly!

**Comment by akhill10** [comment]
@charlesBochet Okay, Actually I was looking for the bounty from Algora as I was the first one to claim it. But never mind 😞

**Comment by charlesBochet** [comment]
Sorry about that @akhill10, I did not checked the algora dashboard, I will do next time

**Pull request closed without merging**